### ISSUE TITLE
Phpmyadmin works very slowly with large number or tables

### ISSUE THREAD

**Issue by amsharma9** [phpmyadmin/phpmyadmin#16188]
Hi,

We have Phpmyadmin installed on AWS C5.xlarge with 8Gb RAM, 4 vCPUs running Ubuntu 16.0.4 . We have around forty thousand tables in our database. Storage engine is Innodb.  Phpmyadmin takes a long time to open (most probably because it is trying to get info about all tables from `information_schema` database). It takes about:

- 2.5 to 3 minutes to open the left panel after clicking database name.
- 3.5 to 4 minutes after that to get table names on clicking `tables`.
- 1 minute to Select a database
- Average 1 minute to execute any SQL in the SQL tab

How can we speed this up. Can reduce it search etc. We are fine with not seeing any navigation lists etc. The slow speed is hampering our work. After reading on internet we have already set the following in the cfg file:
```
$cfg['MaxExactCount'] = 0;
$cfg['MaxExactCountViews'] = 0;
```

Easily reproducible. We also have some third party tools which show that Phpmyadmin pages are the slowest to execute and we are being questioned on our use of Phpmyadmin. If you need access to check, kindly let me know.

### Server configuration
- Operating system: Ubuntu 16.0.4
- Web server: Apache 2.4
- Database version: MySQL 5.7
- PHP version: 7.3
- phpMyAdmin version: 

### Client configuration
- Browser: Brave Browser (Chrome)
- Operating system: Windows 10

Regards
Amal

**Comment by williamdes** [comment]
Hi @amsharma9 
This is an excellent question !
Your feedback will be very precious, I started a similar topic on #16005 

Is there any chance that I would have some access to develop improvements ?

Also let me know if the next version that is still in development, available on our website [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/) is better or worse in performance.

**Comment by amsharma9** [comment]
Yes, sure. These statistics are from a Production server but we can always share the Staging server details if you need or implement some changes you do, on this server and give you feedback on them.

We would be glad to help you improve your product. It will benefit us all, right!!!
Will install 5.1. We installed it using apt-get and assumed it's the latest. You have so many versions after that, totally missed it.

**Comment by williamdes** [comment]
> Yes, sure. These statistics are from a Production server but we can always share the Staging server details if you need or implement some changes you do, on this server and give you feedback on them.
> 
> We would be glad to help you improve your product. It will benefit us all, right!!!
> Will install 5.1. We installed it using apt-get and assumed it's the latest. You have so many versions after that, totally missed it.

Sure, feel free to email me at williamdes@wdes.fr to share the details 

About apt get you will have 4.9 available 5.0 is planned but requires a lot of work for the packaging team (https://github.com/phpmyadmin/phpmyadmin/issues/15236#issuecomment-615954993 for install instructions)

**Comment by rajat315315** [comment]
Same happening here.
I can confirm on master branch. Although the speed is faster than on 4.6.x.
We have around 250 tables.
On 4.6.x, to open left panel, it took around 1 minute.
On master or 5.x, it took lesser, not sure about exact seconds.

**Comment by rajat315315** [comment]
I think @amsharma9 can use cache to quickly see the list of tables. Setting `query_cache_type` to 1 turns it on.
As it would affect in general all results, you might not want to do it.

**Comment by amsharma9** [comment]
Thanks @rajat315315. Sorry for the delayed reply. **Is this for MySQL and not Phpmyadmin specific?** I just checked, this is deprecated in MySQL 5.7.20 and removed in 8.0.3. We are on MySQL 5.7.23 and planning to move to MySQL 8 in next week or two. Do you think it will work?

**Comment by williamdes** [comment]
Let us know if the performances did increase with MySQL 8, I hope so :)

NB: I did not have some time to write benchmarks for you but I still did not forget to do it someday

**Comment by shucon** [comment]
Hey @williamdes , I'm just thinking some solutions but would using Local storage be helpful here? We'll have to manage deleted and newly added tables and local storage limitations, but it would make things faster after the first query.
Or we could start by preventing querying of all table names of a database at once.(better IMO)

**Comment by williamdes** [comment]
> Hey @williamdes , I'm just thinking some solutions but would using Local storage be helpful here? We'll have to manage deleted and newly added tables and local storage limitations, but it would make things faster after the first query.
> Or we could start by preventing querying of all table names of a database at once.(better IMO)

Hey, I am not sure we should rely on browser features to fix such performance issues.
Finding the root cause would be a better idea.

**Comment by shucon** [comment]
Okay, let me know if you want me to work on it.

**Comment by williamdes** [comment]
Hi @amsharma9 

Is there any way you could install the pecl module `xhprof` on your server and let me provide you a script so you can give me some data back ?

**Comment by amsharma9** [comment]
@williamdes Yes, sure. Just to update you, we haven't moved to MySQL 8.x as there were other priorities. We have also reduced total tables to about twenty thousand by using some settings to avoid creation of unused tables on setup of each sub-site. We have about seven thousand sub-sites.

**Comment by williamdes** [comment]
> @williamdes Yes, sure. Just to update you, we haven't moved to MySQL 8.x as there were other priorities. We have also reduced total tables to about twenty thousand by using some settings to avoid creation of unused tables on setup of each sub-site. We have about seven thousand sub-sites.

I understand, good work 👏
Let me know when you have the extension installed

**Comment by williamdes** [comment]
Hi @amsharma9
Do you have some news?

**Comment by amsharma9** [comment]
Hi @williamdes, sorry for the delay in responding. My colleagues mentioned there was some compatibility issue installing xhprof but we have finally been able to install it on our staging today. Will do it on "production" tonight. Let us know what you want us to do. I do need some input from you - does it impact performance?

FYI, In the meantime we also upgraded our production server to MySQL 8.0.21 - in case that fact changes anything for you.

**Comment by williamdes** [comment]
Hi @amsharma9 

> I do need some input from you - does it impact performance?

No, this is why I like hxprof :)

> Let us know what you want us to do.

- First I would like you to install the the latest version in development [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/#devel)
- Second you will need to follow the last steps [here](https://github.com/phpmyadmin/phpmyadmin/wiki/How-to-use-xhprof#patch-your-indexphp-file)
- And then just send me profiling files to williamdes@wdes.fr (just in case they contain some private data, normally they do not)

Let me know if that works for you ;)

**Comment by acedigibits** [comment]
any update on this issue ? i have over 18000 tables and tables do not stop processing.

**Comment by williamdes** [comment]
> any update on this issue ? i have over 18000 tables and tables do not stop processing.

Would you mind following https://github.com/phpmyadmin/phpmyadmin/wiki/How-to-use-xhprof and sending me a profile please ?
